### PR TITLE
(#2051520) top-level drop-in support

### PR DESCRIPTION
**Pull request by jacekmigacz**

### CODE REVIEW

**Comment by mrc0mmand** [comment]
After a quick glance I think we might want to backport 7a670b1dd981c645064f69faf85b04620aadbafb as well (didn't have time yet to test if it applies to RHEL 8 build as well, but given the code looks the same it probably does).

File: src/core/service.c [hunk 1]
```diff
@@ -536,24 +536,9 @@ static int service_verify(Service *s) {
...
-                                return -ENOEXEC;
-                        }
-                        if (strv_isempty(command->argv)) {
-                                log_unit_error(UNIT(s),
-                                               "Service has an empty argv in %s=. Refusing.",
-                                               service_exec_command_to_string(c));
-                                return -ENOEXEC;
-                        }
-                }
+        if (!service_unit_name_is_valid(UNIT(s)->id)) {
```
Comment (dtardon) [inline src/core/service.c]: This hunk is wrong: it should only add a new check.

**Review by dtardon — CHANGES_REQUESTED** [review]
The 1st commit erroneously removes existing code. Also, the 2nd and 3rd commits refer to invalid upstream commits.

File: src/core/service.c (hunk 1 above)
Comment (jacekmigacz) [reply src/core/service.c]: Hunk remains in place, while that new check was removed in 3e1db806b0c18fd6138886ce67fac2655f09caef.

**Review by dtardon — APPROVED** [review]
LGTM

**Review by mrc0mmand — APPROVED** [review]
Also LGTM (and qa_ack'ed the respective BZ), thanks!

**Pull request merged**